### DISCUSSION TITLE
Copilot only produces broken code... At least with Kotlin

### DISCUSSION THREAD

**Discussion by gnarea** [community/community#50105] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I've been playing with Copilot on [veraid-jvm](https://github.com/relaycorp/veraid-jvm), and I'm yet to get a **working** suggestion. It only produces broken code.

Here's an example:

I was starting to define the body of `SignatureMetadata.encode()` when I got this suggestion:

![copilot1](https://user-images.githubusercontent.com/369863/225282888-3f4550f5-3f9e-4979-a427-3c172ec2a309.png)

Now, I know for a fact that `DatePeriod` doesn't have an `encode()` method because I just defined that class. Nevertheless, I accepted it, hoping I might just need to tweak it a bit, but it turns out that the constructor of`Attribute` takes 1 or 2 arguments (not 3 as Copilot suggested).

![copilot3](https://user-images.githubusercontent.com/369863/225283905-8f9505dd-6add-4013-8eb8-8745cea0dd0e.png)

This is just one example. I've got around 5 suggestions so far and they're all wrong. It's more of a productivity killer than a productivity booster... Which is particularly bad considering Kotlin is statically typed.

**Comment by gnarea** [comment]
OK, I've been using it for just over an hour now, and it's got a lot better. I wonder if it just needs some kind of _priming_ internally when used on a project for the first time. Most suggestions (not all) are now spot on.

**Reply by abdullzz** [reply]
had to dislike this one, this issue persist for most of us
